dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java

```diff
@@ -139,7 +141,11 @@ private void setMsgInUrl(String msg) {
             if (!url.contains(URL_SPLICE_CHAR)) {
                 type = URL_SPLICE_CHAR;
             }
-            url = String.format("%s%s%s=%s", url, type, contentField, msg);
+            try {
+                url = String.format("%s%s%s=%s", url, type, contentField, URLEncoder.encode(msg, DEFAULT_CHARSET));
```

Comment:
Hi, @Radeity @fuchanghai , If add this, Should I remove the 124 lines of code to construct the URL, and then directly construct the URI through url?